airflow/models/xcom.py

```diff
@@ -458,8 +469,8 @@ def clear(
         return query.delete()
 
     @staticmethod
-    def serialize_value(value: Any):
-        """Serialize Xcom value to str or pickled object"""
+    def serialize_value(value: Any, key=None, task_id=None, dag_id=None, execution_date=None, run_id=None):
```

Comment:
We shouldn't have execution_date on main anymore, and we should add map_index now.
```suggestion
    def serialize_value(*, value: Any, key: str, task_id: str, dag_id: str, run_id: str, map_index: int = -1):
```

Reply:
updated. any reason you put the `*` before value and not after?

Reply:
To make them all keyword-only arguments

Reply:
ha, yes of course.  i was thinking in comparison with [here](https://github.com/apache/airflow/blob/ddb5246bd1576e2ce6abf8c80c3328d7d71a75ce/airflow/models/xcom.py#L101) but upon closer inspection that's just an overload method and not the actual signature. and i also had in mind the question of signature change -- making `value` kwarg only where previously it was not. but i guess since we're  changing signature anyway it may not matter.

Reply:
hmm... yeah it does break some tests to make `value` keyword-only

```
tests/models/test_xcom.py::TestXCom::test_resolve_xcom_class_fallback_to_basexcom: TypeError: serialize_value() takes 0 positional arguments but 1 was given
tests/models/test_xcom.py::TestXCom::test_resolve_xcom_class_fallback_to_basexcom_no_config: TypeError: serialize_value() takes 0 positional arguments but 1 was given
Error: Process completed with exit code 1.
```

e.g. here's one test:
```python
    @conf_vars({("core", "xcom_backend"): "", ("core", "enable_xcom_pickling"): "False"})
    def test_resolve_xcom_class_fallback_to_basexcom(self):
        cls = resolve_xcom_backend()
        assert issubclass(cls, BaseXCom)

        assert cls().serialize_value([1]) == b"[1]"
```

it seems unlikely that a user would be relying on this.   but it's possible.  for now i've updated it to keep `value` as positional -- but please do let me know your thoughhts.